bittensor/utils/__init__.py

```diff
@@ -68,22 +68,28 @@ def unbiased_topk(values, k, dim=0, sorted=True, largest=True):
     return topk, permutation[indices]
 
 
-def version_checking():
-    response = requests.get(bittensor.__pipaddress__)
-    latest_version = response.json()["info"]["version"]
-    version_split = latest_version.split(".")
-    latest_version_as_int = (
-        (100 * int(version_split[0]))
-        + (10 * int(version_split[1]))
-        + (1 * int(version_split[2]))
-    )
+def version_checking(timeout: int = 15):
+    try:
+        response = requests.get(bittensor.__pipaddress__, timeout=timeout)
+        latest_version = response.json()["info"]["version"]
+        version_split = latest_version.split(".")
+        latest_version_as_int = (
+            (100 * int(version_split[0]))
+            + (10 * int(version_split[1]))
+            + (1 * int(version_split[2]))
+        )
 
-    if latest_version_as_int > bittensor.__version_as_int__:
-        print(
-            "\u001b[33mBittensor Version: Current {}/Latest {}\nPlease update to the latest version at your earliest convenience\u001b[0m".format(
-                bittensor.__version__, latest_version
+        if latest_version_as_int > bittensor.__version_as_int__:
+            print(
+                "\u001b[33mBittensor Version: Current {}/Latest {}\nPlease update to the latest version at your earliest convenience\u001b[0m".format(
+                    bittensor.__version__, latest_version
+                )
             )
-        )
+
+    except requests.exceptions.Timeout:
+        bittensor.logging.error("Version check failed due to timeout")
+    except requests.exceptions.RequestException as e:
+        bittensor.logging.error(f"Version check failed due to request failure: {e}")
```

Comment:
what about logging the url of the request so people can see manually the bittensor version?

Reply:
1cee00fa1724fd00463d48ba84eb367e0e8c57ac Think it makes sense to log upfront that way if we error there aren't duplicated logs for each error.